functional_testing/Openvcloud/ovc_master_hosted/OVC/c_advanced/machine_tests.py

```diff
@@ -73,20 +74,52 @@ def test003_check_connectivity_through_external_network(self):
         #. Check if you can ping VM1 from outside, should succeed
         """
 
-    @unittest.skip('Not Implemented')
     def test004_migrate_vm_in_middle_of_writing_file(self):
-        """ OVC-000
+        """ OVC-039
         *Test case for checking data integrity after migrating vm in the middle of writing a file*
 
         **Test Scenario:**
 
         #. Create a cloudspace CS1, should succeed
-        #. Create VM1
+        #. Create VM1, should succeed
         #. Write a big file FS1 on VM1
         #. Migrate VM1 in the middle of writing a file, should succeed
         #. Check if the file has been written correctly after vm live migration
         """
-        # Note: this testcase may be hard to be implemented from here.
+        self.lg('%s STARTED' % self._testID)
+
+        self.lg("Create VM1,should succeed.")
+        vm1_id = self.cloudapi_create_machine(cloudspace_id=self.cloudspace_id)
+        self.assertTrue(vm1_id)
```

Comment:
no need for this line, `cloudapi_create_machine` method already assert for the machine id
https://github.com/0-complexity/G8_testing/blob/dd1c6dc70c64eefdae6ce9916c5d0535efd3ae23/functional_testing/Openvcloud/utils/utils.py#L225

Reply:
done